### PR TITLE
dnsdist: Move dnsdist specific files to `dnsdistdist`, format and delint them

### LINKED ISSUE #12242: dnsdist: move and format dnsdist specific files to dnsdistdist
- Program: Authoritative, Recursor, dnsdist 
 - Issue type: Bug report

### Environment

 - Operating system: 
 - Software version: 
 - Software source: 

### Steps to reproduce

1. 
2. 
3.

### PR DESCRIPTION
**Pull request by rgacogne**
### Short description

This PR does:
- remove symbolic links for dnsdist-specific files from `pdns/dnsdistdist/`
- move these files to `pdns/dnsdistdist/`
- format the moved files
- then all the moved files are delinted, one file at a time to make it easier to review the changes

Closes #12242 

### Checklist

I have:
- [x] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [x] compiled this code
- [x] tested this code
- [ ] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [x] added or modified unit test(s)

### CODE REVIEW

**Review by Habbie — APPROVED** [review]
the refactors look good. All the small delints look straightforward. I did not read every single line. Approved!

**Comment by omoerbeek** [comment]
Compile error on macOS:
`dnsdist.cc:1754:186: error: use of undeclared identifier 'mmsghdr'` I suppose some `#ifdef` got lost

Also there's one symlink of a dnsdist related file remaining: `dnsdist.conf-dist -> ../dnsdistconf.lua`

**Review by omoerbeek — APPROVED** [review]
Build and unit tested on OpenBSD and macOS. Did not check every delint, but what I checked looked good.

File: pdns/dnsdistdist/dnsdist-lua-bindings.cc [hunk 1]
```diff
@@ -419,34 +420,34 @@ void setupLuaBindings(LuaContext& luaCtx, bool client, bool configCheck)
...
-  luaCtx.registerFunction<std::string (ClientState::*)() const>("getEffectiveTLSProvider", [](const ClientState& fe) {
+  luaCtx.registerFunction<std::string (ClientState::*)() const>("getEffectiveTLSProvider", [](const ClientState& frontend) {
     setLuaNoSideEffect();
-    if (fe.tlsFrontend != nullptr) {
-      return fe.tlsFrontend->getEffectiveProvider();
+    if (frontend.tlsFrontend != nullptr) {
+      return frontend.tlsFrontend->getEffectiveProvider();
     }
-    else if (fe.dohFrontend != nullptr) {
+    if (frontend.dohFrontend != nullptr) {
```
Comment (chbruyand) [inline pdns/dnsdistdist/dnsdist-lua-bindings.cc]: Or doqFrontend ?

Comment (rgacogne) [reply pdns/dnsdistdist/dnsdist-lua-bindings.cc]: GH is having a hard time telling me exactly which line you are commenting on, but I think this is related to the TLS providers, ie `getConfiguredTLSProvider` or `getConfiguredTLSProvider`? For DoQ it's a bit tricky, I guess it should be `BoringSSL`?

Comment (chbruyand) [reply pdns/dnsdistdist/dnsdist-lua-bindings.cc]: At least GH shows the correct line on my view. But yes, this is about `getConfiguredTLSProvider` and `getConfiguredTLSProvider`.
I agree.

**Pull request merged**